### DISCUSSION TITLE
How to use nginx cache control for my service

### DISCUSSION THREAD

**Discussion by erkanisuf** [cdk8s-team/cdk8s#1696] [Q&A]
Hello.
I have a public map tile service in my cluster and I need to cache most of the responses because the service can't handle many requests due to having heavy data. I couldn't find any example of nginx with cdk8s how to achieve this.

My service is configured as this:

>    new Deployment(scope, `${config.prefix}${config.name}-app`, {
      metadata: {
        name: `${config.prefix}${config.name}`,
        labels: {
          "app.kubernetes.io/name": `${config.prefix}${config.name}`,
        }
      },
      select: true,
      podMetadata: {
        labels: {
          "app.kubernetes.io/name": `${config.prefix}${config.name}`,
        }
      },
      replicas: config.replicas,
      containers: [{
        image: `${imageNamePrefix}${config.name}:${process.env.ARGOCD_APP_REVISION ?? 'latest'}`,
        envFrom: [
          Env.fromSecret(
            Secret.fromSecretName(scope, `${config.prefix}${config.name}-service-env`, `${config.prefix}${config.name}-env`)
          ),
        ],
        securityContext: {
          user: 1000,
          group: 1000,
          readOnlyRootFilesystem: false,
        },
        resources: config.resources,
        readiness: config.readiness,
      }],
    })

Basically I am trying to figure out how I can achieve this: 
[https://github.com/maplibre/martin/blob/main/docs/src/run-with-nginx.md](url)
with cdk8s
